clustered/client/src/main/java/org/ehcache/clustered/client/internal/store/ServerStoreProxy.java

```diff
@@ -44,25 +48,50 @@ public String getCacheId() {
 
   @Override
   public Chain get(long key) {
-    // TODO: Implement ServerStoreProxyImpl.get
-    throw new UnsupportedOperationException("ServerStoreProxyImpl.get not implemented");
+    EhcacheEntityResponse response;
+    try {
+      response = entity.invoke(EhcacheEntityMessage.getOperation(cacheId, key), false);
+    } catch (Exception e) {
+      throw new ServerStoreProxyException(e);
+    }
+    if (response != null && response.getType() == EhcacheEntityResponse.Type.GET_RESPONSE) {
+      return ((EhcacheEntityResponse.GetResponse)response).getChain();
+    } else {
+      throw new ServerStoreProxyException("Response for get operation was invalid.");
```

Comment:
More details please: at least include the type of response you got in the exception's message.